ReactViewControl/ReactViewRender.cs

```diff
@@ -188,16 +189,14 @@ private enum LoadStatus {
             ExecuteLoaderFunction("loadStyleSheet", frameName, loadArg);
         }
 
-        private string GetMappings() {
-            return JavascriptSerializer.Serialize(Plugins.Select(m => new KeyValuePair<string, object>(m.Name, NormalizeUrl(ToFullUrl(m.MainSource)))));
-        }
-
         private void InternalLoadPlugins(string frameName) {
             var pluginsWithNativeObject = Plugins.Where(p => !string.IsNullOrEmpty(p.NativeObjectName)).ToArray();
             var loadArgs = new[] {
-                JavascriptSerializer.Serialize(pluginsWithNativeObject.Select(m => new[] { m.Name, GetNativeObjectFullName(m.NativeObjectName, frameName) })), // plugins
-                JavascriptSerializer.Serialize(Plugins.Select(m => new KeyValuePair<string, object>(m.Name, ToFullUrl(VirtualPathUtility.GetDirectory(NormalizeUrl(m.MainSource)))))),
-                GetMappings()
+                JavascriptSerializer.Serialize(pluginsWithNativeObject.Select(m => new [] {
+                    m.Name,
+                    GetNativeObjectFullName(m.NativeObjectName, frameName),
+                    ToFullUrl(m.MainSource.Replace("\\", ResourceUrl.PathSeparator)),
```

Comment:
NormalizeUrl does the Replace("\\", ResourceUrl.PathSeparator) job

Reply:
But removes the extension as well. Isn't that an issue?

Reply:
That was an "hack" because of require

Reply:
Btw there's an use of "NormalizeUrl" that is later used to set the component cache key. I think that it is working properly even with the extension, so I'll keep it that way